Wikipedia/Code/WMFExploreCollectionViewController.m

```diff
@@ -1081,12 +1081,13 @@ - (nonnull UICollectionReusableView *)collectionView:(UICollectionView *)collect
             return [collectionView dequeueReusableSupplementaryViewOfKind:UICollectionElementKindSectionFooter withReuseIdentifier:WMFFeedEmptyHeaderFooterReuseIdentifier forIndexPath:indexPath];
         case WMFFeedMoreTypeLocationAuthorization: {
             WMFTitledExploreSectionFooter *footer = (id)[collectionView dequeueReusableSupplementaryViewOfKind:UICollectionElementKindSectionFooter withReuseIdentifier:[WMFTitledExploreSectionFooter wmf_nibName] forIndexPath:indexPath];
-
-            for (UIGestureRecognizer *gr in footer.gestureRecognizers) {
-                [footer removeGestureRecognizer:gr];
+            
+            for (UIGestureRecognizer *gr in footer.enableLocationButton.gestureRecognizers) {
```

Comment:
Generally the tap recognizers are for adding gesture recognition to plain views that aren't subclasses of `UIControl` (like `UIButton`). With controls, you can programatically add target objects that respond to the actions of the control - `addTarget:action:forControlEvents:` or connect them with Interface Builder via an `IBAction`.

In this case, it's probably easiest to remove & add the target, similar to how the header button works (line 981ish):

```
        [header.rightButton removeTarget:self action:@selector(headerRightButtonPressed:) forControlEvents:UIControlEventTouchUpInside]; // ensures the view controller isn't duplicated in the target list, causing duplicate actions to be sent
        header.rightButton.tag = indexPath.section;
        [header.rightButton addTarget:self action:@selector(headerRightButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
```

Alternatively, you could add the `IBAction` to the view class, and a delegate protocol & property that could be set up to relay actions back to the view controller (Similar to how `WMFFindInPageKeyboardBar` works). However, since this is only one button and action, the method above is probably better.